### DISCUSSION TITLE
[Tests] Setup StandardRB for Ruby

### DISCUSSION THREAD

**Discussion by kalimar** [Terrastories/terrastories#727] [Temporarily Archived Tickets]
We are going to setup [StandardRB](https://github.com/testdouble/standard) as part of our development environment. This will involve some additional documentation so developers know how to integrate the tool into their text editors.
https://github.com/testdouble/standard#how-do-i-run-standard-in-my-editor

Before merging this we should come to a basic agreement on configuration and local setup.

- [ ] Setup standard https://github.com/testdouble/standard
- [ ] Add documentation for developers to integrate with text editor
- [ ] In separate commits, run the autoformatter to autofix any potential issues.
- [ ] In separate commits, fix individual lint violations that cannot be autocorrected. Flag a steward if you have questions about how to go about this.
- [ ] Ensure tests run and that a click through of the app, both admin and otherwise, still work as expected.

**Comment by rgenchev** [comment]
Can I work on this? Thanks in advance! :)

**Comment by midhunkrishna** [comment]
@kalimar @rgenchev, 

I think the following plan of action should work: 
1. Install rubocop like, in development and test groups

```ruby
gem "rubocop-github"
gem "rubocop-performance"
gem "rubocop-rails"
```

2. Generate config by:

```bash
rubocop --auto-gen-config
```

This should create a .rubocop.yml file and .rubocop-todo.yml, which can be split into multiple tickets and worked on later without creating a single large PR (s ince this project is small and in the early stages, we can still go ahead and do an auto fix, wdyt @kalimar  ) 

3. I would say the default configuration provided would suffice?

4. Editor specific setup instructions are provided in the rubocop documentation itself (https://rubocop.readthedocs.io/en/latest/integration_with_other_tools/#editor-integration). We can include this link to the readme.

**Comment by midhunkrishna** [comment]
Shouldn't we also have a section in [here](https://github.com/Terrastories/terrastories/blob/master/SETUP.md#development) about the different tools that we use for dev?

Eg: 
### We Use:
- rubocop for ruby linting and for ruby styleguide
- jest as a js test runner
- simplecov for code coverage metric

@kalimar @MxOliver @mirandawang

**Comment by rgenchev** [comment]
I created https://github.com/Terrastories/terrastories/pull/340. I wasn't sure if I should update the readme.

**Comment by midhunkrishna** [comment]
@rgenchev I wasn't sure about the readme either. You can update readme like this in this PR -> https://github.com/Terrastories/terrastories/pull/338/files#diff-550856cf67eb7260aaeb883550043af8R143-R149

**Comment by rgenchev** [comment]
@midhunkrishna, I did it here: https://github.com/Terrastories/terrastories/pull/340/commits/2a2d32c3373f467e8a2679fff49cb2833d13dd9c

**Comment by lauramosher** [comment]
Thoughts on utilizing [StandardRB](https://github.com/testdouble/standard)? It is essentially a wrapper around RuboCop, but defines "standards" for which rules are enforced rather than having to determine which ones we want to and why for each cop.

**Comment by rgenchev** [comment]
I could continue working on this. We just have to decide whether we pick `rubocop` or `StandardRB`.

**Comment by rgenchev** [comment]
@lauramosher, @midhunkrishna, @kalimar, what do you think? :)

**Comment by lauramosher** [comment]
@rgenchev HI! Thank you for your interest. I've updated the issue to reflect the decision the stewards have landed on for our linter and formatter. Let us know if you have any questions.

I've assigned you this issue. If you've changed your mind, please let us know and we'll unassign you. Thanks!
